netket/operator/_local_operator.py

```diff
@@ -247,13 +250,13 @@ def __init__(
             operators = [operators]
             acting_on = [acting_on]
 
-        operators = [np.asarray(operator) for operator in operators]
+        operators = [csr_matrix(operator) for operator in operators]
 
         # If we asked for a specific dtype, enforce it.
         if dtype is None:
             dtype = np.promote_types(operators[0].dtype, np.float32)
             for op in operators[1:]:
-                np.promote_types(dtype, op.dtype)
+                dtype = np.promote_types(dtype, op.dtype)
```

Comment:
This was a bug. Can you add a test case checking that this does not happen again (construct a localoperator with two operators, the first being real and the latter complec, and check the resulting dtype)

Reply:
This is the same bug spotted by @femtobit in #915 Which implementation do we leave? I like @femtobit's. Do I add the test here or in #915

Reply:
You can [cherry-pick](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/cherry-picking-a-commit) my commit https://github.com/netket/netket/pull/915/commits/6a747d19273f27fa3736e7c6641d233cfa5ea29f into your branch for this PR if you want, it's independent of the rest of the PR (which is not really compatible with your changes, so I'd wait until this PR here is finished before doing something with #915.)

Reply:
Don't know how to cherry pick because I'm working on a forked repo and I can only fetch the upstream for main. I will just copy paste your code

Reply:
That is ok.

I mainly just wanted a test.
T he code is already ok

Reply:
Hi  @PhilipVinc sorry for answering so late. I was quite busy last week. The test is done

Reply:
thanks!